test/Altinn.AccessManagement.Tests/Controllers/V2RightsInternalControllerTest.cs

```diff
@@ -0,0 +1,156 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Altinn.AccessManagement.Controllers;
+using Altinn.AccessManagement.Core.Models;
+using Altinn.AccessManagement.Models;
+using Altinn.AccessManagement.Tests.Fixtures;
+using Altinn.AccessManagement.Tests.Scenarios;
+using Altinn.AccessManagement.Tests.Seeds;
+using Xunit;
+
+namespace Altinn.AccessManagement.Tests.Controllers;
+
+/// <summary>
+/// <see cref="RightsInternalController"/>
+/// </summary>
+public class V2RightsInternalControllerTest(WebApplicationFixture fixture) : ControllerTest(fixture)
+{
+    private static Action<AcceptanceCriteriaTest> WithAssertDbLastDelegationToUserIsRevoked(IParty from, IUserProfile to) => test =>
+    {
+        test.ApiAssertions.Add(async fixture =>
+        {
+            var delegations = await fixture.Postgres.ListDelegationsChanges(filter => filter.Where(delegation => from.Party.PartyId == delegation.OfferedByPartyId && to.UserProfile.UserId == delegation.CoveredByUserId));
+
+            Assert.True(
+                delegations.OrderBy(delegation => delegation.Created).First().DelegationChangeType == DelegationChangeType.RevokeLast,
+                $"Last delegation from party '{from.Party.Name}' with party ID '{from.Party.PartyId}' to user '{to.UserProfile.Party.Name}' with user ID '{to.UserProfile.UserId}' is not of type '{DelegationChangeType.RevokeLast}'");
+        });
+    };
+
+    private static void WithAssertDbDelegationsNotEmpty(AcceptanceCriteriaTest test)
+    {
+        test.ApiAssertions.Add(async api =>
+        {
+            var delegations = await api.Postgres.ListDelegationsChanges();
+            Assert.NotEmpty(delegations);
+        });
+    }
+
+    private static void WithAssertEmptyDelegationList(AcceptanceCriteriaTest test)
+    {
+        test.ResponseAssertions.Add(async response =>
+        {
+            var delegations = await response.Content.ReadFromJsonAsync<IEnumerable<RightDelegationExternal>>();
+            Assert.Empty(delegations);
+        });
+    }
+
+    private static Action<AcceptanceCriteriaTest> WithAssertResponseContainsDelegationToUserProfile(IParty from, IUserProfile to) => test =>
+    {
+        test.ResponseAssertions.Add(async response =>
+        {
+            var delegations = await response.Content.ReadFromJsonAsync<IEnumerable<RightDelegationExternal>>();
+            var result = delegations.Any(delegation =>
+                    delegation.To.Any(attribute => attribute.Value == from.Party.PartyId.ToString()) &&
+                    delegation.From.Any(attribute => attribute.Value == to.UserProfile.UserId.ToString()));
+
```

Comment:
From/To mixup? 
delegation.To ... from.Party
delegation.From ... to.UserProfile

Reply:
true, nice!